alpaca/rest.go

```diff
@@ -742,6 +744,121 @@ func (c *Client) GetAsset(symbol string) (*Asset, error) {
 	return &asset, nil
 }
 
+type GetOptionContractsRequest struct {
+	UnderlyingSymbols string           `json:"underlying_symbols"`
+	ShowDeliverable   *bool            `json:"show_deliverables"`
+	Status            *OptionStatus    `json:"status"`
+	ExpirationDate    *civil.Date      `json:"expiration_date"`
+	ExpirationDateGTE *civil.Date      `json:"expiration_date_gte"`
+	ExpirationDateLTE *civil.Date      `json:"expiration_date_lte"`
+	RootSymbol        *string          `json:"root_symbol"`
+	Type              *OptionType      `json:"type"`
+	Style             *OptionStyle     `json:"style"`
+	StrikePriceGTE    *decimal.Decimal `json:"strike_price_gte"`
+	StrikePriceLTE    *decimal.Decimal `json:"strike_price_lte"`
+	Limit             *int             `json:"limit"`
+	PPInd             *bool            `json:"ppind"`
+}
+
+// GetOptionContracts returns the list of Option Contracts.
+func (c *Client) GetOptionContracts(req GetOptionContractsRequest) ([]OptionContract, error) {
+	u, err := url.Parse(fmt.Sprintf("%s/%s/options/contracts", c.opts.BaseURL, apiVersion))
+	if err != nil {
+		return nil, err
+	}
+
+	q := u.Query()
+
+	structType := reflect.TypeOf(req)
+	structValue := reflect.ValueOf(req)
+
+	for i := 0; i < structType.NumField(); i++ {
+		fieldType := structType.Field(i)
+		fieldValue := structValue.Field(i)
+
+		kind := fieldType.Type.Kind()
+
+		if kind == reflect.String {
+			v := fieldValue.String()
+			if v != "" {
+				q.Set(fieldType.Tag.Get("json"), v)
+			}
+			continue
+		}
+
+		// else it's a reflect.Ptr:
+		if kind != reflect.Ptr {
+			panic("GetOptionContractsRequest's struct was changed this should be a pointer")
+		}
+
+		// if it's not a nil value then we do somthing with it
+		if !fieldValue.IsNil() {
+			subFieldType := fieldType.Type.Elem()
+			subFieldValue := fieldValue.Elem()
+
+			// again we only really care about the string because it could be empty
+			if subFieldType.Kind() == reflect.String {
+				v := subFieldValue.String()
+				if v != "" {
+					q.Set(fieldType.Tag.Get("json"), v)
+				}
+			} else {
+				q.Set(fieldType.Tag.Get("json"), fmt.Sprint(subFieldValue))
+			}
+		}
+	}
```

Comment:
Not sure if this reflection-based solution would work with my proposed zero-value instead of pointer approach. However, honestly I prefer writing "empty check" for every single field.